jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java

```diff
@@ -2353,4 +2349,28 @@ private void prepare()
             }
         }
     }
+
+    private class ParsedHttpField extends HttpField
+    {
+        public ParsedHttpField(HttpHeader header, String name, String value)
+        {
+            super(header, name, value);
+        }
+
+        @Override
+        protected QuotedCSV newQuotedCSV(String value)
+        {
+            return new QuotedCSV(false, value)
+            {
+                @Override
+                protected void onComplianceViolation(ComplianceViolation violation)
+                {
+                    if (_complianceMode.allows(WHITESPACE_IN_PARAMETER))
+                        _handler.onViolation(new ComplianceViolation.Event(_complianceMode, WHITESPACE_IN_PARAMETER, getValue()));
+                    else
+                        throw new BadMessageException("Bad white space");
```

Comment:
Do we need `BadMessageException` here, or we can just call `super`?

Reply:
We need a `BadMessageException` here so on the server it is turned into a 400 response rather than a 500 response.  +1 on the message.